src/cocaine-app/planner/__init__.py

```diff
@@ -136,19 +137,91 @@ def _ttl_cleanup_planner(self):
                 self.ttl_cleanup_timer.next(),
                 self._ttl_cleanup_planner)
 
+    def _get_idle_groups(self, days_of_idle):
+        """
+        Iterates all over couples. Find couples where ttl_cleanup hasn't run for more then 'period'
+        :param days_of_idle: how long the group could be idle
+        :return: list of groups[0] from couples
+        """
+
+        idle_groups = []
+
+        # the epoch time when executed jobs are considered meaningful
+        idleness_threshold = time.time() - datetime.timedelta(days=days_of_idle).total_seconds()
+
+        cursor = self.collection.find().sort('cleanup_ts', pymongo.ASCENDING)
+        if cursor.count() < len(storage.replicas_groupsets):
+            logger.info('Sync cleanup data is required: {0} records/{1} couples'.format(
+                cursor.count(), len(storage.replicas_groupsets)))
+            self.sync_historic_data(recover_ts=0, cleanup_ts=int(time.time()))
+            cursor = self.collection.find().sort('cleanup_ts', pymongo.ASCENDING)
+
+        READ_MONGO_CHUNK = 4096
+        for i in range(0, cursor.count(), READ_MONGO_CHUNK):
```

Comment:
Fixed in https://github.com/yandex/mastermind/pull/366/commits/4cc3dfb4378c8308c3b781c6d8a4c0b731d2a38d